src/System.Net.Http/tests/FunctionalTests/HuffmanDecodingTests.cs

```diff
@@ -0,0 +1,444 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public class HuffmanDecodingTests
+    {
+        delegate int DecodeDelegate(ReadOnlySpan<byte> src, Span<byte> dst);
+
+        private static readonly DecodeDelegate s_decodeDelegate = GetDecodeDelegate();
+
+        private static DecodeDelegate GetDecodeDelegate()
+        {
+            Assembly assembly = typeof(HttpClient).Assembly;
+            Type huffmanType = assembly.GetType("System.Net.Http.HPack.Huffman");
+            MethodInfo decodeMethod = huffmanType.GetMethod("Decode", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
+            return (DecodeDelegate)Delegate.CreateDelegate(typeof(DecodeDelegate), decodeMethod);
```

Comment:
Not going to block on it, but I don't like that we have to do this.

Reply:
I think it's the best of not-so-great options.